Sources/PackageLoading/Diagnostics.swift

```diff
@@ -40,6 +40,10 @@ extension Basics.Diagnostic {
         return .warning("ignoring duplicate product '\(product.name)'\(typeString)")
     }
 
+    static func duplicateProduct(productID: String) -> Self {
+        return .warning("ignoring duplicate product '\(productID)'")
```

Comment:
I think here too we should split up identity and name for the diagnostic, similar to existing `duplicateTargetDependency()`.